zerver/views/custom_profile_fields.py

```diff
@@ -132,21 +132,9 @@ def remove_user_custom_profile_data(request: HttpRequest, user_profile: UserProf
                                     data: List[int]=REQ(validator=check_list(
                                                         check_int))) -> HttpResponse:
     for field_id in data:
-        try:
-            field = CustomProfileField.objects.get(realm=user_profile.realm, id=field_id)
-        except CustomProfileField.DoesNotExist:
-            return json_error(_('Field id {id} not found.').format(id=field_id))
-
-        try:
-            field_value = CustomProfileFieldValue.objects.get(field=field, user_profile=user_profile)
-        except CustomProfileFieldValue.DoesNotExist:
-            continue
-        field_value.delete()
-        notify_user_update_custom_profile_data(user_profile, {'id': field_id,
-                                                              'value': None,
-                                                              'rendered_value': None,
-                                                              'type': field.field_type})
-
+        error = do_check_remove_custom_profile_field_value(user_profile, field_id)
+        if error:
+            return error
```

Comment:
If you do the changes I suggest above, you get to remove the whole `error = / if error` blocks.